### DISCUSSION TITLE
How to Preprocess meshopt_simplifySloppy

### DISCUSSION THREAD

**Discussion by Abdelilah-Majid** [zeux/meshoptimizer#387] [Q&A] [answered]
hey so i have this game engine that i want to implement LOD in it
and i want to use meshopt_simplifySloppy 
and event that this function is very fast but when i run it against 50 models the frame rate decreases from 70 fps to 4 fps
and i was thinking about ways to improve this like SIMD and multithreading but i thought that even if i did them it will still take a lot of frame time from the cpu so one idea was to preprocess the calculations and what i mean by that is to calculate all of what the function meshopt_simplifySloppy need to calculate at time when the model is loaded and then store the result in an array and then when i need to do LOD(lets say when the camera move) i can use this array to reduce the number of triangles of the model; so this array will be like a sorted index array which is sorted from the indices that need to be remove first to the indices that need to be removed last
so is there is a function that does this in your library and if not how can i do this(if you dont know how to do it please just explane the code in the function meshopt_simplifySloppy so that i can figure out a way to do it my self) and i am not asking to write the code for what i want i just need like a guide of the possible way to do it

thanks in advance;

**Comment by zeux** [comment, marked as answer]
I don't see how this is a bug. Please use Discussions for questions. You can of course simplify ahead of time, see https://github.com/zeux/meshoptimizer/blob/master/demo/main.cpp#L508 for a suggested sequence (you can replace simplify with simplifySloppy there).

**Reply by Abdelilah-Majid** [reply]
sorry for labeling this as a bug; the reason i made this as a issue is because i have opened a  discussion a while ago about another thing and no one reply for it; again i am sorry; i will take a look at the code that you pointed at and i will reply to you soon; and thank you for replying;

**Comment by Abdelilah-Majid** [comment]
that was exactly what i was looking for but in a different way; i was looking for a sorted indices array but this will do the trick thank you very much; i will try to do it with 30 or so levels
